### DISCUSSION TITLE
generateUserImports with call to mountDomRenderer.

### DISCUSSION THREAD

**Discussion by TJTorola** [react-cosmos/react-cosmos#1654] [Ideas]
I'm not sure I understand why there is the extra hoop to jump through of adding a file with:

```
import { mountDomRenderer } from 'react-cosmos-dom';
import * as mountArgs from './cosmos.imports';
 
mountDomRenderer(mountArgs);
```

When instead `cosmos.imports` could do that call to mountDomRenderer on it's own and we could just compile that. This hasn't been that big a deal to me, but in pursuit of trying to keep my source directory clean, I started writing a plugin for my personal project (which uses esbuild) and I ran into a bit of a roadblock because of this. I end up needing to write two virtual files, one which is that documented file, that imports the src generated by generateUserImports. There is probably a way to do this with esbuild (I saw how it is done with the vite plugin). But it would be made much easier if instead generateUserImports or some other similar function could generate all the source that I need to compile.

Otherwise, great app, thank you all for building it, it has been a joy to use.

**Comment by ovidiuch** [comment]
@TJTorola Ideally you wouldn't need to see/know about the `cosmos.imports` module and the compiler would abstract it away like it does with the Vite and Webpack integrations.

The [Custom Bundler](https://reactcosmos.org/docs/getting-started/custom-bundler) instructions are a DYI backup strategy that doesn't make any compiler assumptions. The basic idea is to write a minimal renderer index module that imports a dynamically generated module map, which could in theory work without any compiler—just pure ESM.

I haven't had time to create an ESBuild plugin/abstraction but if you've put something together feel free to share it. Would be great to offer an ESBuild integration as part of the Getting Started guides.

**Comment by TJTorola** [comment]
@ovidiuch That's fair, I got something working, but it is largely based on my own codebase, I would need to spend time figuring out the previously mentioned issue of how to have a esbuild `stdin` file import another `stdin` file before having anything worth sharing. If I come up with something I'll make sure to share it though.
